src/substrait/textplan/converter/InitialPlanProtoVisitor.cpp

```diff
@@ -629,9 +630,24 @@ void InitialPlanProtoVisitor::updateLocalSchema(
   // Revamp the output based on the output mapping if present.
   auto mapping = getOutputMapping(relation);
   if (!mapping.empty()) {
+    // TODO -- Use a more explicit check.
     if (!relationData->outputFieldReferences.empty()) {
```

Comment:
What sort of more explicit check were you thinking?

```suggestion
    if (!relationData->outputFieldReferences.empty()) {
      assert(relationData->relation.rel_type_case() == ::substrait::proto::Rel::kAggregate));
```

I'm not familiar enough with substrait to know at what level aggregate nodes are the only relations with output field references. Do you mean they are the only ones supported by substrait-cpp? The only relation for which textplan will ever provide a representation? I don't *think* they're the only relations which substrait itself allows to have an output mapping...

I think the comment should explain this in greater depth. If it's the last thing (IE, part of general substrait knowledge) maybe a link to an appropriate section of the substrait doc is in order. 

In any case, if this block of code is specific to `relationData->relation.rel_type_case() == ::substrait::proto::Rel::kAggregate`, it should probably be moved into the switch above

Reply:
This was relation into the comment below.  I've replaced the output field references with a check on the incoming proto's type.  I did not move it above as the function is currently separated into a section where the fields are populated with a later emit handling step.  I can see some potential refactoring where those two parts of this rather long function could be split up.